### ISSUE TITLE
Get rid of unused CPI service 

### ISSUE THREAD

**Issue by lubronzhan** [vmware-tanzu/community-edition#2619]
## Bug Report

Currently, the CPI manifest contains a Service CR which is no longer used by upstream anymore.
https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/vsphere-cpi/1.22.3/bundle/config/upstream/vsphere-cpi/04-service.yaml

This could cause conflicts when other NodePort service use this port.

## Expected Behavior

No service created for CPI

## Steps to Reproduce the Bug

Create another NodePort service with port 43001 before creating CPI

## Screenshots or additional information and context
Checking the pods under kube-system namespace, we could see that vsphere cloud provider is restarted for multi times. 
```
root@harbor:~[56330]# kubectl --context workload-113-admin@workload-113 get pods -n kube-system | grep cloud
vsphere-cloud-controller-manager-6gl6t 0/1 CrashLoopBackOff 52 4h18m
```
Checking its details log, it is showing error of `"F0721 05:07:22.260682 1 server.go:114] Server Listen() failed: listen tcp :43001: bind: address already in us".` That is the reason of workload cluster hits createStalled.

```
0721 05:07:22.260342 1 clientconn.go:882] blockingPicker: the picked transport is not ready, loop back to repick
I0721 05:07:22.259953 1 reflector.go:175] Starting reflector *v1.Node (0s) from k8s.io/client-go@v0.0.0-20201201102650-cc926379659d/tools/cache/reflector.go:125
F0721 05:07:22.260682 1 server.go:114] Server Listen() failed: listen tcp :43001: bind: address already in use
goroutine 166 [running]:
k8s.io/klog/v2.stacks(0xc0000be001, 0xc00071a540, 0x75, 0xba)
 /go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1026 +0xb8
k8s.io/klog/v2.(*loggingT).output(0x3ccea00, 0xc000000003, 0x0, 0x0, 0xc000896380, 0x3bed792, 0x9, 0x72, 0x0)
 /go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:975 +0x1a3
k8s.io/klog/v2.(*loggingT).printf(0x3ccea00, 0xc000000003, 0x0, 0x0, 0x0, 0x0, 0x24f740f, 0x1a, 0xc0003b7a70, 0x1, ...)
 /go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:750 +0x18c
k8s.io/klog/v2.Fatalf(...)
 /go/pkg/mod/k8s.io/klog/v2@v2.4.0/klog.go:1502
k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/server.(*server).Start.func1(0xc000992870)
 /go/src/github.com/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/server/server.go:114 +0x13f
created by k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/server.(*server).Start
 /go/src/github.com/k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/server/server.go:111 +0x54
```
## Environment Details

* Build version (`tanzu version`): 1.5 and before
* Deployment (Managed/Standalone cluster): 
* Infrastructure Provider (Docker/AWS/Azure/vSphere): 
* Operating System (client):

**Comment by joshrosso** [comment]
Hey @vmware-tanzu/pkg-vsphere-cpi-owners! Are you able to take a look at this and see if the service is no longer needed? And if so, prioritize its change?

**Comment by lubronzhan** [comment]
Yeah I'm one of the owner, we'll fix this in 1.23

**Comment by lubronzhan** [comment]
This can be closed
